### PR TITLE
Remove service instance register and deregister cmd

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/istio/istio/issues/29153

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: releasenotes/notes/deprecate-register-deregister-cmd.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: deprecate-istioctl-cmd
+area: istioctl
+issue:
+  - 29153
+
+releaseNotes:
+  - |
+    **Deprecated** istioctl register and deregister command
```
Comment (shamsher31) [inline releasenotes/notes/deprecate-register-deregister-cmd.yaml:9]:
```suggestion
    **Removed** istioctl register and deregister commands.
```

File: releasenotes/notes/deprecate-register-deregister-cmd.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: deprecate-istioctl-cmd
```
Comment (shamsher31) [inline releasenotes/notes/deprecate-register-deregister-cmd.yaml:2]: same as https://github.com/istio/istio/pull/29593#discussion_r542111625

Comment (carolynhu) [reply releasenotes/notes/deprecate-register-deregister-cmd.yaml:2]: yes, thanks @shamsher31! I also aware of that, will update the word to removed

File: releasenotes/notes/remove-register-deregister-cmd.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+issue:
+  - 29153
+
+releaseNotes:
+  - |
+    **Removed** istioctl register and deregister command
```
Comment (shamsher31) [inline releasenotes/notes/remove-register-deregister-cmd.yaml:9]:
```suggestion
    **Removed** istioctl register and deregister commands.
```

**Review by shamsher31 — APPROVED** [review]
Thanks.

File: releasenotes/notes/remove-register-deregister-cmd.yaml (hunk 3 above)
Comment (shamsher31) [reply releasenotes/notes/remove-register-deregister-cmd.yaml:9]: @carolynhu you can remove `hold` after this is fixed.

**Review by howardjohn — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Pull request merged**